platform/src/server/http/ssl_config.ts

```diff
@@ -64,10 +67,35 @@ export class SslConfig {
     this.enabled = config.enabled;
     this.key = config.key;
     this.certificate = config.certificate;
-    this.certificateAuthorities = config.certificateAuthorities;
+    this.certificateAuthorities = this.initCertificateAuthorities(config.certificateAuthorities);
     this.keyPassphrase = config.keyPassphrase;
-    this.keystore = config.keystore;
     this.cipherSuites = config.cipherSuites;
     this.supportedProtocols = config.supportedProtocols;
   }
+
+  /**
+   * Options that affect the OpenSSL protocol behavior via numeric bitmask of the SSL_OP_* options from OpenSSL Options.
+   */
+  getSecureOptions() {
+    if (this.supportedProtocols === undefined || this.supportedProtocols.length === 0) {
+      return 0;
+    }
+
+    const supportedProtocols = this.supportedProtocols;
+    return Array.from(protocolMap).reduce((secureOptions, [protocolAlias, secureOption]) => {
+      // `secureOption` is the option that turns *off* support for a particular protocol,
+      // so if protocol is supported, we should not enable this option.
+      return supportedProtocols.includes(protocolAlias)
+        ? secureOptions
+        : secureOptions | secureOption;
```

Comment:
Do we know what this line actually does in a practical sense? Just wondering. We don't see the bitwise or very often.

Reply:
Just [sets the proper bit in bitmask](https://github.com/nodejs/node/blob/eec659c138f3afcd51a25cd8b197d715a47baa65/deps/openssl/openssl/include/openssl/ssl.h#L314-L320) for SSL context options [used by OpenSSL](https://www.openssl.org/docs/man1.0.2/ssl/SSL_CTX_set_options.html)? Or what do you mean?